index.d.ts

```diff
@@ -0,0 +1,65 @@
+declare module 'addsearch-js-client' {
+    export default class AddSearchClient {
+      constructor(sitekey: string, privatekey?: string);
+  
+      search(keyword: string, callback: (results: any) => void): void;
+      search(callback: (results: any) => void): void;
+  
+      suggestions(prefix: string, callback: (suggestions: any) => void): void;
+  
+      autocomplete(field: string, prefix: string, callback: (autocomplete: any) => void): void;
+  
+      fetchCustomApi(field: string, customFilterObject: any, callback: (results: any) => void): void;
+  
+      fetchRangeFacets(options: { field: string, ranges: any[] }, customFilterObject: any, callback: (results: any) => void): void;
+  
+      recommendations(options: any, callback: (recommendations: any) => void): void;
+  
+      getDocument(id: string): Promise<any>;
+      saveDocument(document: any): Promise<any>;
+      saveDocumentsBatch(batch: { documents: any[] }): Promise<any>;
+      deleteDocument(id: string): Promise<any>;
+      deleteDocumentsBatch(batch: { documents: string[] }): Promise<any>;
+  
+      setApiHostname(hostname: string, conf?: { statsApiRequestOnly?: boolean, searchApiRequestOnly?: boolean }): void;
+      getSettings(): any;
+      setLanguage(lang: string): void;
+      setCategoryFilters(categories: string[]): void;
+      addCustomFieldFilter(fieldName: string, value: any): void;
+      removeCustomFieldFilter(fieldName: string, value: any): void;
+      setPriceRangeFilter(minCents: number, maxCents: number): void;
```

Comment:
According to the example in doc, these parameters seem like `string` but not `number`
I haven't tested though, they should be tested.